✈️ Airflow Service (#494)
moj-analytical-services/user-guidance: source/documentation/services/airflow/index.md

## Concepts

We organise Airflow pipelines using **environments**, **projects** and **workflows**:

  * **Environments** are the different stages of infrastructure we provide: `development`, `test` and `production`

  * **Projects** are a unit for grouping workflows dedicated to a distinct business area, business domain, or specific project.
  For example: `BOLD`, `HMCTS` or `HMPPS`.

  * **Workflows** are pipelines, also referred to as [DAGs](https://airflow.apache.org/docs/apache-airflow/2.10.3/core-concepts/dags.html#dags).
  This is a list of the tasks to work through, organised to reflect the relationships between the tasks.
  The workflow definition includes other information such as your repository name and release tag.

## Getting started

You will need to provide us with a container image and a workflow manifest.

The container image will be built and pushed from a GitHub repository you [create](#creating-a-repository) and maintain.

The workflow manifest will be hosted in our [GitHub repository](https://github.com/ministryofjustice/analytical-platform-airflow).

### Request access

> To access the Airflow componentsy, you will need a GitHub account (see our [Quickstart guide](/get-started.html#3-create-github-account)), and to also be part of the `ministryofjustice` GitHub organisation, which you can join by using [this link](https://github.com/orgs/ministryofjustice/sso).

Reviewer: ```suggestion
To access the Airflow components, you will need a GitHub account (see our [Quickstart guide](/get-started.html#3-create-github-account)), and to also be part of the `ministryofjustice` GitHub organisation, which you can join by using [this link](https://github.com/orgs/ministryofjustice/sso).
```
small typo fix

Author: 🤝 https://github.com/moj-analytical-services/user-guidance/pull/494/commits/fd5bf14c3b4e1066027756baa7f0c7d406102bdb

# Airflow

> This documentation is for the new Analytical Platform Airflow service.
>
> For Data Engineering Airflow, please refer to [Data Engineering Airflow](/tools/airflow).

## Overview

[Apache Airflow](https://airflow.apache.org/) is a workflow management platform for data engineering pipelines. We recommend using it for long-running workflows. 

Pipelines are executed on the Analytical Platform's Kubernetes infrastructure and can interact with services such as Amazon Bedrock and Amazon S3.

Our Kubernetes infrastructure is connected to the MoJO Transit Gateway, which connects to:

> * the MoJ Cloud Platform
> * the Modernisation Platform 
> * HMCTS SDP 

Author: ```suggestion
* MoJ Cloud Platform
* MoJ Modernisation Platform 
* HMCTS SDP
```

> **Please note**: Analytical Platform Airflow does not support pipelines that use `BashOperator` or `PythonOperator`.
>
> This is because we run a multi-tenant Airflow service and do not permit running code on the Airflow control plane, which is where non-container operators would run.

## Concepts

We organise Airflow pipelines using **environments**, **projects** and **workflows**:

  * **Environments** are the different stages of infrastructure we provide: `development`, `test` and `production`

  * **Projects** are a unit for grouping workflows dedicated to a distinct business area, business domain, or specific project.
  For example: `BOLD`, `HMCTS` or `HMPPS`.

  * **Workflows** are pipelines, also referred to as [DAGs](https://airflow.apache.org/docs/apache-airflow/2.10.3/core-concepts/dags.html#dags).
  This is a list of the tasks to work through, organised to reflect the relationships between the tasks.
  The workflow definition includes other information such as your repository name and release tag.

> ## Getting started
> 
> You will need to provide us with a container image and a workflow manifest.
> 
> The container image will be built and pushed from a GitHub repository you [create](#creating-a-repository) and maintain.
> 
> The workflow manifest will be hosted in our [GitHub repository](https://github.com/ministryofjustice/analytical-platform-airflow).

Reviewer 2: ```suggestion
## Getting started

Before you can use Airflow, you'll need to:
* [request access](#request-access) on GitHub
* [create a GitHub repository](#create-a-github-repository) 
* [send us a container image](#send-us-a-container-image)
* [send us a workflow manifest](#create-a-workflow-manifest) 

Follow the next steps to get started.
```

Updating other parts to make them consistent...

## Observability Platform

> This functionality is coming soon.

## Runtime templates

We provide repository templates for the supported runtimes:
- [Python](https://github.com/ministryofjustice/analytical-platform-airflow-python-template)
- R (coming soon).

These templates include:

- GitHub Actions workflow to build and scan your container for vulnerabilities with Trivy.
- GitHub Actions workflow to build and test your container's structure.
- GitHub Actions workflow to perform a dependency review of your repository, if it's public.
- GitHub Actions workflow to build and push your container to the Analytical Platform's container registry.
- Dependabot configuration for updating GitHub Actions, Docker, and dependencies such as Pip.

The GitHub Actions workflows call shared workflows we maintain here.

### Vulnerability scanning

The GitHub Actions workflow to build and scan your container for vulnerabilities with Trivy will alert on any CVEs (Common Vulnerabilities and Exposures) marked `HIGH` or `CRITICAL` that have a fix available.

> ### Configuartion testing

Author: I've just installed CSpell 😸

## Getting started

Before you can use Airflow, you'll need to:

- [request Airflow access](#request-airflow-access) 
- [create a GitHub repository](#create-a-github-repository) 
- [create a release](#create-a-release) 
- [create a project and workflow](#create-a-project-and-workflow)

Follow the next steps to get started.

### Request Airflow access

To access the Airflow components, you'll need to:

* have a GitHub account (see our [Quickstart guide](/get-started.html#3-create-github-account)) 
* [join the `ministryofjustice` GitHub organisation](https://github.com/orgs/ministryofjustice/sso)

When you have joined the `ministryofjustice` GitHub organisation, [submit a request for Airflow access](https://github.com/ministryofjustice/data-platform-support/issues/new?template=analytical-platform-airflow-access-request.yml).

After your request is granted, you will be added to a GitHub team that will give you access to our GitHub repository, and AWS environments.

> Our team manually approves requests. After approval it can take up to 3 hours for you to get AWS access.

### Create a GitHub repository

> If you already have a repository you've used for Airflow, you should create a new one. 

Author: I'm not sure about this line
